src/Build/Evaluation/Conditionals/FunctionCallExpressionNode.cs

```diff
@@ -147,11 +158,30 @@ private IEnumerable<string> ExpandArgumentAsFileList(GenericExpressionNode argum
                 argument = FileUtilities.FixFilePath(argument);
             }
 
-            return state.ExpandIntoTaskItems(argument)
-                .Select(i =>
-                    state.EvaluationDirectory != null && !Path.IsPathRooted(i.ItemSpec)
-                        ? Path.GetFullPath(Path.Combine(state.EvaluationDirectory, i.ItemSpec))
-                        : i.ItemSpec);
+
+            IList<TaskItem> expanded = state.ExpandIntoTaskItems(argument);
+            var expandedCount = expanded.Count;
+
+            if (expandedCount == 0)
+            {
+                return null;
+            }
+
+            var list = new List<string>();
```

Comment:
Consider being explicit about the capacity here to optimze the internal array allocations

``` C#
new List<string>(capacity: expandedCount)
```